modules/programs/oh-my-posh.nix

```diff
@@ -0,0 +1,81 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+
+  cfg = config.programs.oh-my-posh;
+
+  configArgument =
+    optionalString (cfg.configFile != null) "--config ${cfg.configFile}";
+
+  themeArgument =
+    optionalString ((cfg.configFile == null) && (cfg.useTheme != null))
+    "--config ${cfg.package}/share/oh-my-posh/themes/${cfg.useTheme}.omp.json";
+
+in {
+  meta.maintainers = [ maintainers.arjan-s ];
+
+  options.programs.oh-my-posh = {
+    enable = mkEnableOption "oh-my-posh";
+
+    package = mkPackageOption pkgs "oh-my-posh" { };
+
+    configFile = mkOption {
```

Comment:
Could this be a structured `settings` option using `pkgs.formats.json`?

Reply:
Note that a "config" is really a "theme" for oh-my-posh. Defining it as a structured option is probably possible, but it would get very big very fast when defining a theme that way. See for example the default theme: https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/themes/default.omp.json
I expect people to want to take one of the official themes and modify them slightly for their use. Having to first rewrite an entire theme into a `pkgs.formats.json` setting is a lot of work...

Reply:
I think if you want to use an existing JSON file then that shouldn't be a problem using something like

``` nix
programs.oh-my-posh.settings = builtins.fromJSON (
  builtins.readFile "${cfg.package}/share/oh-my-posh/themes/${cfg.useTheme}.omp.json")
```

In principle you could do that and then override individual keys within the theme using regular module system overrides.

Reply:
Thank you for your input. I will look into how this needs to be done, as I have no idea at the moment. :)

Reply:
@rycee I updated the module according to your comments. It still works fine when using `useTheme`, but when using a variation of the `fromJSON` line in my local config, I get the error "the string '{theme JSON here}' is not allowed to refer to a store path". Probably this is caused by my lack of deeper understandig of Nix, but perhaps you could help with it?

Reply:
@rycee friendly ping :)

Reply:
@rycee Sorry for bothering you again... I'm still hoping for an answer. :)
If I add the code below to my config, I'm getting the error `the string '{theme JSON here}' is not allowed to refer to a store path`.
```
  programs = {
    oh-my-posh = {
      enable = true;
      useTheme = "space";
      settings = builtins.fromJSON (builtins.readFile "${pkgs.oh-my-posh}/share/oh-my-posh/themes/unicorn.omp.json");
    };
  };
```

Reply:
Okay, I finally figured it out. The correct way to do it is:
```nix
  programs = {
    oh-my-posh = {
      enable = true;
      settings = builtins.fromJSON (builtins.unsafeDiscardStringContext (builtins.readFile "${pkgs.oh-my-posh}/share/oh-my-posh/themes/space.omp.json"));
    };
  };
```